piksi_tools/console/settings_view.py

```diff
@@ -222,33 +182,43 @@ def revert_to_prior_value(self, name, old, new, error_value=SBP_WRITE_STATUS.TIM
                  "   Error Value: {2}")
         invalid_setting_prompt.text = invalid_setting_prompt.text.format(self.name, new, error_value)
         invalid_setting_prompt.run()
+        self.reverting = False
+
+    def _write_value(self, old, new):
+        if (old != new and old is not Undefined and new is not Undefined):
+            if type(self.value) == unicode:
+                self.value = self.value.encode('ascii', 'replace')
+            self.confirmed_set = False
+            res = self.settings.s_api.write(self.section, self.name, new)
+            if res == settings_write_res_e.SETTINGS_WR_OK:
```

Comment:
What if the write response is never received?  In a crowded serial environment or over radio or a lossy link, it is often the case that you don't receive settings write response.  In that case, is the write retried?

Reply:
More elaborate; write is done only once and libsettings blocks waiting for the answer for 0.5 seconds for 5 times, so 2.5 seconds in total, this behavior was inherited from legacy code. If response is not received `SETTINGS_WR_TIMEOUT` is returned.

Reply:
Correction: libsettings waits for the response for 0.5 seconds for 5 times, so 2.5 seconds in total; write request is sent again on each try; this behavior was inherited from legacy code.

If response is not received `SETTINGS_WR_TIMEOUT` is returned.